canon/src/id.rs

```diff
@@ -125,7 +125,7 @@ impl Id {
     pub(crate) fn encoded_len_for_payload_len(payload_len: usize) -> usize {
         let actual_payload = core::cmp::min(payload_len, PAYLOAD_BYTES);
         // version, length and the actual payload length
-        1 + 2 + actual_payload
+        1 + (payload_len as u16).encoded_len() + actual_payload
```

Comment:
We could use the u64 here instead of u16, since we're not saving any bytes anymore by using u16.